### ISSUE TITLE
Proposal peer review

### ISSUE THREAD

**Issue by bkdevart** [INFO-523-F25/final-project-team-2#1]
The following is the peer review of the project proposal by Team 3. The team members that participated in this review are
Carillo, Neyda - @carrilon7
Knox, Brandon - @bkdevart
Stanton, Mike - @mikestantonmusic
- ...
- Describe the goal of the project.
There are 2 main goals for this project. One goal is to discover whether a song's audio features can be used to identify its genre. Another goal is to identify which audio features can help to predict a song's popularity

- Describe the data used or collected.
The data set has a wide range of characteristics about the artist as well as the song for each song including things like tempo, mode (major or minor), loudness and energy. Some of the data values appear to be in the format of an array, so these may need to be broken out into separate features or analyzed using some NLP techniques. The dataset is quite large

- Describe how the research question will be answered, e.g. what approaches / methods will be used.
The goal is to predict the feature (song hotness) as a continuous variable (regression) and also to predict the genre label as a categorical variable (classification). Some techniques to be explored are dimensionality reduction, Random Forest Gradient boosting, Neural Network. Model performance will be evaluated using metrics such as accuracy, F1-score, RMSE, and R² to ensure both the classification and regression models are robust and comparable.

- Is there anything that is unclear from the proposal?
The proposal and goals are clear. Perhaps it may be useful to quantify the goals. Is there an accuracy (or other score metric) threshold that would constitute satisfactory prediction capabilities? Same for the regression aspect - would there be perhaps an r^2 value that would constitute a "strong" algorithm for predicting song popularity?

- Provide constructive feedback on how the team might be able to improve their project.
I'm sure you are planning to address concerns about overfitting (cross validation, etc). It may be helpful to add something about this in the action plan.

- What aspect of this project are you most interested in and would like to see highlighted in the presentation.
I am interested in if the song's popularity can be predicted at all using the features in the dataset. I would think this to be something intangible or highly complex. If the popularity can be predicted it would be interesting to emphasize any insight as to which features are the most important. If the model factors in highly correlated popularity metrics (like artist hotness or familiarity), the insights found might be minimal. When it comes to classifying the genre, thinking of components like tempo, danceability, etc could be similar across multiple genres, which could make it a hard target to hit. There are quite a few genres (as well as fusions), so the model may also have to limit these to a subset of what is available.

- Provide constructive feedback on any issues with file and/or code organization.
Looks good so far. There may be some redundant files in the main folder, but this could be because the repo is still being updated. I am sure you are also going to include some but adding clear visualization plans, such as correlation heatmaps or feature importance plots, would make the analysis more interpretable and visually engaging.

- (Optional) Any further comments or feedback?
